src/Form/MessageSettingsForm.php

```diff
@@ -132,15 +128,15 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $config = $this->config('message.settings');
+    parent::submitForm($form, $form_state);
+
+    $config = $this->configFactory->getEditable('message.settings');
```

Comment:
just to clarify (2) - I'm looking for an example, to see how/where plugins implemented in different modules hold their config.